### ISSUE TITLE
Question: How to make (spac)emacs read .spacemacs.d/init.el at startup?

### ISSUE THREAD

**Issue by tj64** [syl20bnr/spacemacs#5819]
The docs of (most amazing!) spacemacs always talk about .spacemacs file (instead of .emacs) and .spacemacs.d directory (instead of .emacs.d), but I wonder how this is supposed to work?

Since Emacs starts first it looks for .emacs and .emacs.d, and I have actually seen configuration guides that propose to clone the spacemacs repo into an empty .emacs.d. This works of course, but has one big disadvantage for me (as a Vim noob and Spacemacs newbie): I cannot use both, my well configured and well-known former Emacs and the new unconfigured and unknown Spacemacs, in parallel for an extended transition period.

What is the startup trick to make (Spac)emacs actually look into .spacemacs.d/init.el, that enables me to keep on using my plain Emacs (configured in .emacs.d/init.el) side-by-side with it?

**Comment by StreakyCobra** [comment]
> What is the startup trick to make (Spac)emacs actually look into .spacemacs.d/init.el, that enables me to keep on using my plain Emacs (configured in .emacs.d/init.el) side-by-side with it?

https://github.com/syl20bnr/spacemacs/blob/develop/doc/FAQ.org#try-spacemacs-without-modifying-my-existing-emacs-configuration

:smiley:

**Comment by tj64** [comment]
Perfect, thanks a lot! 

Since ".emacs.d/" is the hardcoded user-emacs-directory in subr.el, I found the quickstart info for spacemacs a bit confusing. It reads as if creating a "~/.spacemacs.d/" (or a simple "~/.spacemacs") would magically convince Emacs to use these for startup, or as if some shellscript in the background would tweak the startup logic.
   
And since Emacs startup logic is not so trivial, nor is spacemacs initialization logic, it would take quite some digging to find out how all this is related.

The workaround is great, but maybe a few explanations about ".spacemacs vs .emacs" and ".spacemacs.d" vs. ".emacs.d" in the quickstart docs could speed up the (for me not so) quick start?

I'm still confused about ".spacemacs.d". The workaround from the FAQ uses .emacs.d. Replacing existing Emacs config works by cloning the spacemacs repo into .emacs.d. So thats still the user-emacs-directory, thats were spacemacs config is located - where does .spacemacs.d enter the game?

**Comment by StreakyCobra** [comment]
Spacemacs is just a configuration on top of Emacs, so when you start Emacs, Emacs first uses one of the following configurations files:

- `.emacs` 
- `.emacs.d/init.el`

as for any emacs configuration.

In order to use Spacemacs, you have to clone the whole configuration in `.emacs.d`, and the `init.el` is the entry point that load Spacemacs in Emacs. Note that if you have an `.emacs` file, it will be used first and the Spacemacs' `.emacs.d/init.el` will not be loaded (except if you load it manually in the `.emacs`).

The following files are handled by Spacemacs *only*, i.e. Spacemacs has to be loaded first through `.emacs.d/init.el`, and will later read the following files for adapting Spacemacs to users preferences:

- `.spacemacs `
- `.spacemacs.d/init.el`

So in other words: you can only have **one** configuration of Emacs: either yours of Spacemacs. If you use the spacemacs one, you can additionally use and customize one of the Spacemacs configuration file.

**Comment by tj64** [comment]
Thanks for the explanations. So the .spacemacs.d and its init.el file are entirely different beasts from their .emacs.d counterparts, therefore my confusion. 

From reading the quickstart info I concluded that a .spacemacs.d, if present, will be assigned as new user-emacs-directory, and Emacs will use its init.el for startup. But startup is like always, only that spacemacs delivers the init.el instead of myself, and I can put some extra config in the dotspacemacs file or directory.

**Comment by StreakyCobra** [comment]
Exactly :+1:

If you have some idea to make this clearer, feel free to propose it in a PR :-)

**Comment by tj64** [comment]
Fabien Dubosson <notifications@github.com> writes:

> Exactly 
> :+1:
>
> If you have some idea to make this clearer, feel free to propose it in
> a PR :-)

Done, but I'm not sure what to do with the attached patch?
Do you have commit rights?



-- 
cheers,
Thorsten

**Comment by StreakyCobra** [comment]
Patches don't go through Github mailing lists I think. Contributions are done throught Github's PR. We have a [guide](https://github.com/syl20bnr/spacemacs/wiki/Beginner's-Guide-to-Contributing-a-Pull-Request-to-Spacemacs) explaining the steps, and be sure to have a look at our [contrubutions guideline](https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org) first :smiley:

**Comment by tj64** [comment]
Puh, I didn't read that ... ;-(
